### ISSUE TITLE
Search with address bar, search last entry instead when hit enter

### ISSUE THREAD

**Issue by marekbujko** [brave/brave-browser#34944]
## Description
If I leave something (say “ABC”) in the address bar in tab A, go to another tab B, go back to tab A and modify it, say change it to “DEF”, and hit enter, it will search “ABC” instead of “DEF”. It doesn’t matter what search engine I use. Doesn’t matter how you modify it (Delete the whole string and type new one, or add something at the end), it just searches the original.
Putting website is the same as well. Put google.com first, then come back and change it to yahoo.com 1 but only goes to google.com

Same as:
https://community.brave.com/t/search-with-address-bar-search-last-entry-instead-when-hit-enter/519770

## Steps to Reproduce

   1. Type anything in the address bar in tab A, “ABC”
   2. Go to another tab
   3. Go back to tab A
   4. Change the query in the address bar in tab B, “DEF”
   5. Hit “Enter”
   6. Search results for “ABC” is shown

## Reproduces how often: 
Always

## Brave version
Brave	1.61.104 Chromium: 120.0.6099.115

## Version/Channel Information:
Current release
Windows 10 22H2 latest and Ubuntu 20.04.6 LTS latest

**Comment by rebron** [comment]
@marekbujko Do you have any extensions that are on? Also what search engine do you have by default? You're saying you can't change it?

**Comment by MadhaviSeelam** [comment]
Tried to reproduce the issue in following scenarios in Win 11 x64
```
Brave | 1.61.104 Chromium: 120.0.6099.115 (Official Build) (64-bit)
-- | --
Revision | 38497f14ca9ff6f1d5b9c1d72f61fec6f0efcf1d
OS | Windows 11 Version 22H2 (Build 22621.2861)
```

Scenarios 1:  `Show autocomplete suggestions in address bar` toggle `on` by default in `brave://settings/appearance`
 - Could not reproduce.

Scenario 2:  `Show autocomplete suggestions in address bar` toggled `off` 
  - Reproduced the issue 
  
Scenario 3:  `Show autocomplete suggestions in address bar` toggle `on` but unchecked `Top sites`, `Browsing History`, `Bookmarks`, `Leo AI Assistant`
  - Could not reproduce
  
https://github.com/brave/brave-browser/assets/98358127/516cd383-7009-4d7d-9f1b-b59f1e2c6a83
 
 cc: @rebron

**Comment by marekbujko** [comment]
> Tried to reproduce the issue in following scenarios in Win 11 x64
> 
> ```
> Brave | 1.61.104 Chromium: 120.0.6099.115 (Official Build) (64-bit)
> -- | --
> Revision | 38497f14ca9ff6f1d5b9c1d72f61fec6f0efcf1d
> OS | Windows 11 Version 22H2 (Build 22621.2861)
> ```
> 
> Scenarios 1: `Show autocomplete suggestions in address bar` toggle `on` by default in `brave://settings/appearance`
> 
> * Could not reproduce.
> 
> Scenario 2: `Show autocomplete suggestions in address bar` toggled `off`
> 
> * Reproduced the issue
> 
> Scenario 3: `Show autocomplete suggestions in address bar` toggle `on` but unchecked `Top sites`, `Browsing History`, `Bookmarks`, `Leo AI Assistant`
> 
> * Could not reproduce
> 
>  2023-12-18_12h07_56.mp4 
> cc: @rebron

Yes, Scenario 2. I have set **Show autocomplete suggestions in address bar** toggled **off**.

**Comment by marekbujko** [comment]
> @marekbujko Do you have any extensions that are on? Also what search engine do you have by default? You're saying you can't change it?
I have set **Show autocomplete suggestions in address bar** toggled **off**.
If I set to **on**, then it works without problems.
